Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py

```diff
@@ -55,79 +168,215 @@ def highjack_transition_rando():
+    total_con_left += 2
+
+
+def link_list_to_transitions(
+    link_list: list[tuple[Area, Area]],
+    transitions_map: MutableMapping[Transition, Transition],
+    origins_bucket: list[Transition],
+    redirections_bucket: list[Transition],
+):
+    for link in link_list:
+        options_original = [
+            trans for trans in origins_bucket
+            if trans.from_ == link[0].area_id
+        ]
+        options_redirect = [
+            trans for trans in redirections_bucket
+            if trans.to == link[1].area_id
+        ]
+        original = random.choice(options_original)
+        redirect = random.choice(options_redirect)
+        transitions_map[original] = redirect
+        origins_bucket.remove(original)
+        redirections_bucket.remove(redirect)
+
+        counterpart_original = Transition(from_=redirect.to, to=redirect.from_)
+        counterpart_redirect = Transition(from_=original.to, to=original.from_)
+        transitions_map[counterpart_original] = counterpart_redirect
+        origins_bucket.remove(counterpart_original)
+        redirections_bucket.remove(counterpart_redirect)
+
+
+def get_random_redirection(original: Transition, all_redirections: Iterable[Transition]):
+    possible_redirections = [
+        redirect for redirect in all_redirections
+        if original.from_ != redirect.to  # Prevent looping on itself
+    ]
+    if len(possible_redirections) > 0:
+        return random.choice(possible_redirections)
+    return None
 
 
 def set_transitions_map():
-    _possible_transitions_bucket = list(starmap(Transition, ALL_POSSIBLE_TRANSITIONS))
-    """A temporary container of transitions to pick from until it is empty."""
     transitions_map.clear()
-    for area in TRANSITION_INFOS_DICT.values():
-        for to_og in (exit_.area_id for exit_ in area.exits):
-            original = Transition(from_=area.area_id, to=to_og)
-            redirect = get_random_redirection(original, _possible_transitions_bucket)
-            if redirect is None or original in transitions_map:
-                # Don't override something set in a previous iteration,
-                # like from linked two-way entrances.
+    if not CONFIGS.SKIP_JAGUAR:
+        starting_default = TRANSITION_INFOS_DICT[starting_area].default_entrance
+        tutorial_original = Transition(from_=LevelCRC.JAGUAR, to=LevelCRC.PLANE_CUTSCENE)
+        tutorial_redirect = Transition(from_=starting_default, to=starting_area)
+        transitions_map[tutorial_original] = tutorial_redirect
+
+    _possible_redirections_bucket = list(starmap(Transition, ALL_POSSIBLE_TRANSITIONS))
+    one_way_list = [
+        Transition(from_=LevelCRC.WHITE_VALLEY, to=LevelCRC.MOUNTAIN_SLED_RUN),
+        Transition(from_=LevelCRC.MOUNTAIN_SLED_RUN, to=LevelCRC.APU_ILLAPU_SHRINE),
+        Transition(from_=LevelCRC.APU_ILLAPU_SHRINE, to=LevelCRC.WHITE_VALLEY),
+        Transition(from_=LevelCRC.CAVERN_LAKE, to=LevelCRC.JUNGLE_CANYON),
+    ]
```

Comment:
This is duplicated from one of the sections in `disabled_exits`. Let's extract that.

Or even better, encoding one-way transitions/exits as part of `transition_infos`. (that can be done later, let's at least de-dupe that information for now)

Reply:
This one_way_list is only defined once now, so no more duplication.